api/server/httputils/errors.go

```diff
@@ -66,6 +66,7 @@ func GetHTTPErrorStatusCode(err error) int {
 			{"this node", http.StatusServiceUnavailable},
 			{"needs to be unlocked", http.StatusServiceUnavailable},
 			{"certificates have expired", http.StatusServiceUnavailable},
+			{"repository does not exist", http.StatusNotFound},
```

Comment:
Please don't add this here. This is a really bad pattern we're trying to remove.

I think the right way to handle this is to have `TranslatePullError()` return a specific error, and then translate that error into a `StatusNotFound` in the api/server/router/ function for pull